src/BinSkim.Driver/AnalyzeCommand.cs

```diff
@@ -261,58 +270,114 @@ private HashSet<IBinarySkimmer> CreateSkimmers(IMessageLogger<BinaryAnalyzerCont
+                                RuleUtilities.BuildRuleDisabledDueToMissingPolicyMessage(
+                                    context.Rule.Name,
+                                    reasonForNotAnalyzing));
+                            disabledSkimmers.Add(skimmer.Id);
+                            break;
+                        }
 
                         case AnalysisApplicability.ApplicableToSpecifiedTarget:
+                        {
+                            try
                             {
-                                try
-                                {
-                                    skimmer.Analyze(context);
-                                }
-                                catch (Exception ex)
-                                {
-                                    LogUnhandledRuleExceptionAnalyzingTarget(disabledSkimmers, context, skimmer, ex);
-                                }
-                                break;
+                                skimmer.Analyze(context);
                             }
+                            catch (Exception ex)
+                            {
+                                LogUnhandledRuleExceptionAnalyzingTarget(disabledSkimmers, context, skimmer, ex);
+                            }
+                            break;
+                        }
                     }
                 }
-                DisposeContext(context);
+
+                // Once we've processed all portable executable skimmers for a specific
+                // target, we'll proactively let go of the data associated with this 
+                // analysis phase. Follow-on analyses (such as the Roslyn integration)
+                // shouldn't attempt to rehydrate this data. The context implementation
+                // currently raises an exception if there is an attempt to rehydrate a
+                // previously nulled PE instance.
+                DisposePortableExecutableContextData(context);
+
+                // IsManagedAssembly is computed on intitializing the binary context
+                // object and is still valid after disposing the PE data. The Roslyn
+                // analysis is driven solely off the binary file path in the context.
+                if (context.IsManagedAssembly && roslynAnalyzerFilePaths?.Count > 0)
+                {
+                    AnalyzeManagedAssembly(context.Uri.LocalPath, roslynAnalyzerFilePaths, context);
+                }
+            }
+        }
+
+        private static void DisposePortableExecutableContextData(BinaryAnalyzerContext context)
+        {
+            context.DisposePortableExecutableData();
+        }
+
+        private void AnalyzeManagedAssembly(string assemblyFilePath, IEnumerable<string> roslynAnalyzerFilePaths, BinaryAnalyzerContext context)
+        {
+            ILDiagnosticsAnalyzer roslynAnalyzer = new ILDiagnosticsAnalyzer();
+
+            foreach (string analyzerFilePath in roslynAnalyzerFilePaths)
```

Comment:
Note that we reload all analyzers for each analysis target, which entails reflecting over the analyzer set again and again. This could be made more efficient.